tests/integrations/mcs/tso/keyspace_group_manager_test.go

```diff
@@ -122,32 +126,143 @@ func (suite *tsoKeyspaceGroupManagerTestSuite) TestTSOKeyspaceGroupSplit() {
+	re.NoError(err)
+	re.NotNil(tam)
+	return tam.HandleRequest(tsopkg.GlobalDCLocation, count)
+}
+
+func (suite *tsoKeyspaceGroupManagerTestSuite) TestTSOKeyspaceGroupSplitElection() {
+	re := suite.Require()
+	// Create the keyspace group 1 with keyspaces [111, 222, 333].
+	handlersutil.MustCreateKeyspaceGroup(re, suite.pdLeaderServer, &handlers.CreateKeyspaceGroupParams{
+		KeyspaceGroups: []*endpoint.KeyspaceGroup{
+			{
+				ID:        1,
+				UserKind:  endpoint.Standard.String(),
+				Members:   suite.tsoCluster.GetKeyspaceGroupMember(),
+				Keyspaces: []uint32{111, 222, 333},
+			},
 		},
-		DcLocation: tsopkg.GlobalDCLocation,
-		Count:      count,
-	}
-	tsoClient, err := client.Tso(ctx)
+	})
+	kg1 := handlersutil.MustLoadKeyspaceGroupByID(re, suite.pdLeaderServer, 1)
+	re.Equal(uint32(1), kg1.ID)
+	re.Equal([]uint32{111, 222, 333}, kg1.Keyspaces)
+	re.False(kg1.IsSplitting())
+	// Split the keyspace group 1 to 2.
+	handlersutil.MustSplitKeyspaceGroup(re, suite.pdLeaderServer, 1, &handlers.SplitKeyspaceGroupByIDParams{
+		NewID:     2,
+		Keyspaces: []uint32{222, 333},
+	})
+	kg2 := handlersutil.MustLoadKeyspaceGroupByID(re, suite.pdLeaderServer, 2)
+	re.Equal(uint32(2), kg2.ID)
+	re.Equal([]uint32{222, 333}, kg2.Keyspaces)
+	re.True(kg2.IsSplitTarget())
+	// Check the leadership.
+	member1, err := suite.tsoCluster.WaitForPrimaryServing(re, 111, 1).GetMember(111, 1)
+	re.NoError(err)
+	re.NotNil(member1)
+	member2, err := suite.tsoCluster.WaitForPrimaryServing(re, 222, 2).GetMember(222, 2)
 	re.NoError(err)
-	defer tsoClient.CloseSend()
-	re.NoError(tsoClient.Send(req))
-	return tsoClient.Recv()
+	re.NotNil(member2)
+	// Wait for the leader of the keyspace group 1 and 2 to be elected.
+	testutil.Eventually(re, func() bool {
+		return len(member1.GetLeaderListenUrls()) > 0 && len(member2.GetLeaderListenUrls()) > 0
+	})
+	// Check if the leader of the keyspace group 1 and 2 are the same.
+	re.Equal(member1.GetLeaderListenUrls(), member2.GetLeaderListenUrls())
+	// Resign and block the leader of the keyspace group 1 from being elected.
+	member1.(*member.Participant).SetCampaignChecker(func(*election.Leadership) bool {
+		return false
+	})
+	member1.ResetLeader()
+	// The leader of the keyspace group 2 should be resigned also.
+	testutil.Eventually(re, func() bool {
+		return member2.IsLeader() == false
+	})
+	// Check if the leader of the keyspace group 1 and 2 are the same again.
+	member1.(*member.Participant).SetCampaignChecker(nil)
+	testutil.Eventually(re, func() bool {
+		return len(member1.GetLeaderListenUrls()) > 0 && len(member2.GetLeaderListenUrls()) > 0
+	})
+	re.Equal(member1.GetLeaderListenUrls(), member2.GetLeaderListenUrls())
+	// Finish the split.
+	handlersutil.MustFinishSplitKeyspaceGroup(re, suite.pdLeaderServer, 2)
+}
+
+func (suite *tsoKeyspaceGroupManagerTestSuite) TestTSOKeyspaceGroupSplitClient() {
+	// TODO: remove the skip after the client is able to support multi-keyspace-group.
+	suite.T().SkipNow()
```

Comment:
it's clever. You can check in first. Then I can remove the skip here so that I can enable this test as part of #6231